packages/pigeon/lib/swift_generator.dart

```diff
@@ -668,19 +683,15 @@ String _camelCase(String text) {
 }
 
 String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
-  final String forceUnwrap = type.isNullable ? '' : '!';
-  final String castUnwrap = type.isNullable ? '?' : '';
   if (isEnum(root, type)) {
-    final String nullableConditionPrefix =
-        type.isNullable ? '$value == nil ? nil : ' : '';
-    return '$nullableConditionPrefix${_swiftTypeForDartType(type)}(rawValue: $value as! Int)$forceUnwrap';
+    return '${_swiftTypeForDartType(type)}(rawValue: $value as! Int)!';
   } else if (type.baseName == 'Object') {
     // Special-cased to avoid warnings about using 'as' with Any.
     return value;
-  } else if (type.baseName == 'int') {
-    return '($value is Int) ? Int64($value as! Int) : $value as! Int64$castUnwrap';
+  } else if (type.isNullable) {
+    return '(nilOrValue(value: $value) as Any) as! ${_swiftTypeForDartType(type)}?';
```

Comment:
What is the `as Any` for here? Why can't we just cast the nilOrValue result directly?

Reply:
nil can't be cast to `Any`, and `Any?` can't be cast to other types, so it is converted to `Any` first to allow it to be cast to the other types.